enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,376 @@
+  - Not targeted for a future enhancement at this time
+- Provide automated backups for non-self hosted architectures like Hypershift
+
+
+### User Stories
+
+- As a cluster administrator I want to initiate a one-time cluster backup without requiring a root shell on a control plane node so as to minimize the risk involved
+- As a cluster administrator I want to schedule recurring cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes)
+- As a cluster administrator I want to have failure to take cluster backups for more than a configurable period to be reported to me via critical alerts
+
+
+## Proposal
+
+### Workflow Description
+
+#### One time backups
+
+To enable one-time backup requests via an API, a new CRD, `EtcdBackupRequest` (open to better name suggestions), will be used to trigger one-time backup requests.
+
+A new controller in the cluster-etcd-operator, `EtcdBackupRequestController`, will reconcile backup requests as follows:
+- Watch for new `EtcdBackupRequest` CRs as created by an admin
+- Create a backup Job configured for the backup request spec
+- Track the backup progress, failure or success on the `EtcdBackupRequest` status
+
+**TODO:** Decide if `EtcdBackupRequest` needs to be cluster-scoped?
+
+#### Scheduled backups
+
+To enable recurring backups a new cluster-scoped singleton CRD `EtcdBackupSchedule` (open to better name suggestions) will be used to specify the backup schedule, retention policy and other related configuration.
+
+A new controller in the cluster-etcd-operator `EtcdBackupScheduleController` would then reconcile the `EtcdBackupSchedule` CRD with the following workflow:
+- Watches the `EtcdBackupSchedule` CR as created by an admin
+- Creates a [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) that would in turn create `EtcdBackupRequest` CRs at the desired schedule
+- Updates the CronJob for any changes in the schedule
+- Fulfils the specified retention policy by pruning existing backup files before creating a new `EtcdBackupRequest`
+- Prunes completed `EtcdBackupRequest` CRs older than a default time period e.g 24hrs
+  - **TODO:** Should this period be configurable?
+
+
+### API Extensions
+
+#### EtcdBackupSchedule API
+
+The `EtcdBackupSchedule` CRD will be introduced to the API group `config.openshift.io` with the version `v1alpha1`. The CRD would be feature-gated with the annotation `release.openshift.io/feature-set: TechPreviewNoUpgrade` until the prerequisite e2e test has an acceptable pass rate. See the Test Plan and Graduation Criteria sections for more details.
+
+The spec will be as listed below, while the status will be empty since the status of individual backups will be tracked on the `EtcdBackupRequest` status block.
+
+**TODO**: While config CRs don't normally seem to have a status block, where do we track the cumulative status e.g total number of backups saved?
```

Comment:
if the operator is handling retention, something on etcd.operator.openshift.io seems ideal.